contracts/staking/ClusterRewards.sol

```diff
@@ -493,15 +477,14 @@ contract ClusterRewards is
 
     bytes32 public constant RECEIVER_PAYMENTS_MANAGER = keccak256("RECEIVER_PAYMENTS_MANAGER");
 
-    mapping(address => uint256) public receiverBalance;
-    mapping(address => uint256) public receiverRewardPerEpoch;
+    mapping(address => ReceiverPayment) public receiverRewardPayment;
 
-    function _increaseReceiverBalance(address staker, uint256 amount) external override onlyRole(RECEIVER_PAYMENTS_MANAGER) {
-        receiverBalance[staker] += amount;
+    function _increaseReceiverBalance(address staker, uint128 amount) external override onlyRole(RECEIVER_PAYMENTS_MANAGER) {
+        receiverRewardPayment[staker].rewardRemaining += amount;
     }
 
-    function _setReceiverRewardPerEpoch(address staker, uint256 rewardPerEpoch) external override onlyRole(RECEIVER_PAYMENTS_MANAGER) {
+    function _setReceiverRewardPerEpoch(address staker, uint128 rewardPerEpoch) external override onlyRole(RECEIVER_PAYMENTS_MANAGER) {
         require(staker != address(0), "CRW: address 0");
-        receiverRewardPerEpoch[staker] = rewardPerEpoch;
+        receiverRewardPayment[staker].rewardPerEpoch = rewardPerEpoch;
```

Comment:
Add sanity  checks  so that it doesn't go above uint128. Also inputs can be uint256 instead ?

Reply:
can use uint256, but will have to downcast it. 
that will be un-necessary code